rotkehlchen/premium/premium.py

```diff
@@ -234,57 +249,77 @@ def authenticate_device(self) -> None:
-                f'Could not fetch the list of devices due to missing key {e}',
-            ) from e
+            response = self.session.post(
+                url=f'{self.rotki_nest}devices/check',
+                json=self.sign(
+                    method='devices',
+                    device_identifier=device_id,
+                ),
+                headers={'Content-Type': 'application/json'},
+            )
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(f'Failed to check device registration due to: {e}') from e
 
-        device_id = machineid.hashed_id(self.username)
+        if response.status_code == HTTPStatus.FORBIDDEN:
+            raise PremiumAuthenticationError('Premium credentials not valid')
+
+        if response.status_code == HTTPStatus.OK:
+            # Device is already registered
+            return
+
+        if response.status_code == HTTPStatus.NOT_FOUND:
+            # Device is not registered, try to register it
+            self._register_new_device(device_id)
+        else:
+            # Unexpected status code
+            check_response_status_code(
+                response=response,
+                status_codes=[HTTPStatus.OK, HTTPStatus.NOT_FOUND],
+            )
 
-        for device in device_data['devices']:
-            device = cast('dict[str, str]', device)
-            if (remote_id := device.get('device_identifier')) == device_id:
-                break
-            if remote_id is None:
-                log.error(f'Remote device {device} has no identifier in the server response')
-        else:  # device not found
-            if num_devices < devices_limit:
-                # try to register the device
-                self._register_new_device(device_id)
-            else:
-                # user has to edit his devices
-                raise PremiumAuthenticationError(
-                    f'The limit of {devices_limit} devices has been reached',
-                )
-
-    def _register_new_device(self, device_id: str) -> dict:
+    def _register_new_device(self, device_id: str) -> None:
         """
         Register a new device at the rotki server using the provided id.
         May raise:
         - RemoteError
         - PremiumAuthenticationError: if the queried API returns a 401 error
         """
         log.debug(f'Registering new device {device_id}')
-        method = 'devices'
-        device_name = platform.system()
-        data = self.sign(
-            method=method,
-            device_identifier=device_id,
-            device_name=device_name,
-        )
-
         try:
             response = self.session.put(
-                url=f'{self.rotki_api}{method}',
-                data=data,
+                url=f'{self.rotki_nest}{(method := "devices")}',
+                json=self.sign(
+                    method=method,
+                    device_identifier=device_id,
+                    device_name='neto',
```

Comment:
What is this?

Reply:
This is for the devices. We need to assign default names to the devices, probably random ones